### PR TITLE
minimal QR code generator and drawing examples

### PR DESCRIPTION
**Pull request by ladyada**

### CODE REVIEW

File: adafruit_miniqr.py [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+        self.setup_type_info(test, mask_pattern)
+
+        if self.type >= 7:
+            self.setup_type_number(test)
+
+        if self.data_cache is None:
+            self.data_cache = QRCode.create_data(self.type, self.ECC, self.data_list)
+        self.map_data(self.data_cache, mask_pattern)
+
+    def setup_position_probe_pattern(self, row, col):
```
Comment (tannewt) [inline adafruit_miniqr.py]: Should all of these start with _ to be private? This is an interesting API comparison: https://github.com/lincolnloop/python-qrcode

Comment (ladyada) [reply adafruit_miniqr.py]: i dont know if you'd ever wanna use them outside the library but i could do it sure - i ported this, which i think is a port of the above
https://github.com/SHA2017-badge/micropython-examples-sha2017/tree/master/QRcode

Comment (tannewt) [reply adafruit_miniqr.py]: Do you want to change it now or wait? I was thinking its a good change because it reduces the public API complexity.

Comment (ladyada) [reply adafruit_miniqr.py]: yeah ill do it now one minute

Comment (ladyada) [reply adafruit_miniqr.py]: kk see new commit

**Review by tannewt — APPROVED** [review]
Awesome thanks!

**Pull request merged**